src/wallet/rpcwallet.h

```diff
@@ -20,7 +20,7 @@ void RegisterWalletRPCCommands(CRPCTable &t);
  * @param[in] request JSONRPCRequest that wishes to access a wallet
  * @return nullptr if no wallet should be used, or a pointer to the CWallet
  */
-CWallet *GetWalletForJSONRPCRequest(const JSONRPCRequest& request);
+std::shared_ptr<CWallet> GetWalletForJSONRPCRequest(const JSONRPCRequest& request);
```

Comment:
I think this should just return a raw pointer.

Reply:
The RPC handler must *hold* the wallet to prevent concurrent unload.